src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs

```diff
@@ -19,6 +19,7 @@ namespace System.ComponentModel
     /// to the test string itself, since mask literals cannot be modified (i.e: replacing on a literal position
     /// will actually replace on the nearest edit position forward).
     /// </summary>
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] // needed by Clone() to preserve derived ctors
```

Comment:
I see your comment in there but I still don't fully follow. What does having a DynamicallyAccessedMembers attribute on a class mean? Looking at your code I would have expected Clone to instead have a DynamicDependency attribute?

Reply:
See https://github.com/mono/linker/pull/1929.

Maybe we should update the `DynamicallyAccessedMembersAttribute` docs to describe what it means when the attribute is applied to a class/interface/struct?

Reply:
Opened https://github.com/dotnet/runtime/pull/51408 for this.